onnxruntime/python/tools/transformers/shape_infer_helper.py

```diff
@@ -0,0 +1,66 @@
+import os
+import sys
+
+# In ORT Package the symbolic_shape_infer.py is in ../tools
+if os.path.exists("../tools/symbolic_shape_infer.py"):
+    sys.path.append(os.path.join(os.path.dirname(__file__), '../tools'))
+else:
+    sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from symbolic_shape_infer import *
+
+
+class SymbolicShapeInferenceHelper(SymbolicShapeInference):
+    def __init__(
+            self,
+            model,
+            dynamic_axis_mapping={
+                "batch_size": 4,
+                "seq_len": 7
+            },  # use pytorch inputs seq_len as hints
+            verbose=0,
+            int_max=2**31 - 1,
+            auto_merge=True,
+            guess_output_rank=False):
+        super().__init__(int_max, auto_merge, guess_output_rank, verbose)
+        self.__infer(model, dynamic_axis_mapping)
+
+    def __infer(self, model, dynamic_axis_mapping):
+        self.dynamic_axis_mapping_ = dynamic_axis_mapping  # e.g {"batch_size" : 4, "seq_len" :7}
+
+        all_shapes_inferred = False
+        self._preprocess(model)
+        while self.run_:
+            all_shapes_inferred = self._infer_impl()
+
+        if not all_shapes_inferred:
+            raise Exception("Incomplete symbolic shape inference")
```

Comment:
How often will this happen? Could you run a test on huggingface pretrained models in benchmark to make sure it is good for most models.

Reply:
I have tried for one or two model in each category and there is no exception, but let me test for the full list